src/types.tsx

```diff
@@ -0,0 +1,43 @@
+export type BlockContentType =
+  | "header1"
+  | "header2"
+  | "header3"
+  | "paragraph"
+  | "hyperlink"
+  | "numbered_list"
+  | "bullet_list"
+  | "list_item";
+
+export type TextContentType =
+  | "bold"
+  | "italic"
+  | "underline"
+  | "break"
+  | "text";
+
+export type MediaContentType = "image" | "video";
+
+export type NodeType = {
+  type: BlockContentType | TextContentType | MediaContentType;
+  nodes: NodeType[];
+  attributes?: { [key: string]: string };
+  text?: string;
+};
+
+// export type TextNodeType = {
+//   type: TextContentType;
+//   text: string;
+// };
```

Comment:
I know there is a better way to be more explicit based on what attributes are required. Due to the allotted time,  I decided to comment out these types, because I was not sure how to effectively use them.